backend/pick-git/src/main/java/com/woowacourse/pickgit/authentication/presentation/interceptor/IgnoreAuthenticationInterceptor.java

```diff
@@ -0,0 +1,53 @@
+package com.woowacourse.pickgit.authentication.presentation.interceptor;
+
+import com.woowacourse.pickgit.authentication.infrastructure.AuthorizationExtractor;
+import java.util.Objects;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+public class IgnoreAuthenticationInterceptor implements HandlerInterceptor {
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+        Object handler) throws Exception {
+        if (isPreflightRequest(request)) {
+            return true;
+        }
+        if (isGetRequest(request)) {
+            throw new IllegalArgumentException("get 요청일 때만 가능");
+        }
+
+        String authentication = AuthorizationExtractor.extract(request);
+        if (authentication != null) {
+            request.setAttribute("authentication", authentication);
+        }
+        return true;
+    }
+
+    private boolean isPreflightRequest(HttpServletRequest request) {
+        return isOptions(request) && hasAccessControlRequestHeaders(request) && hasAccessControlRequestMethod(request)
+            && hasOrigin(request);
+    }
+
```

Comment:
```suggestion
    private boolean isPreflightRequest(HttpServletRequest request) {
        return isOptions(request)
        && hasAccessControlRequestHeaders(request)
        && hasAccessControlRequestMethod(request)
        && hasOrigin(request);
    }
```

이렇게 개행하면 보기 더 편할 것 같습니다!